java-checks/src/main/java/org/sonar/java/checks/UndocumentedApiCheck.java

```diff
@@ -198,11 +211,26 @@ private String className() {
     return className;
   }
 
-  private WildcardPattern[] getPatterns() {
-    if (patterns == null) {
-      patterns = PatternUtils.createPatterns(forClasses);
+  private WildcardPattern[] getInclusionPatterns() {
+    if (inclusionPatterns == null) {
+      if (StringUtils.isEmpty(forClasses)) {
+        forClasses = "**";
+      }
+
+      inclusionPatterns = PatternUtils.createPatterns(forClasses);
+    }
+    return inclusionPatterns;
+  }
+
+  private WildcardPattern[] getExclusionPatterns() {
+    if (exclusionPatterns == null) {
+      if (StringUtils.isEmpty(exclusion)) {
+        return new WildcardPattern[0];
```

Comment:
Maybe I'm not understanding it right, but in every case the exclusionsPatterns will end up with a non-null value if it was null. If it's either an empty string (maybe isBlank would be better?) or null it will become an empty array, otherwise the pattern will be parsed and the patterns assigned. So the first time this method is called the exlcusionPatterns will be null and after the call it can't be null anymore.